### PR TITLE
support evaluation on vehicleid

### PR DESCRIPTION
**Pull request by zwzhang123**

### CODE REVIEW

**Comment by zwzhang121** [comment]
improve codes for supporting vehicleid

File: openunreid/apis/runner.py [hunk 1]
```diff
@@ -311,7 +314,7 @@ def save(self, mAP=None):
             self._best_mAP = max(self._best_mAP, mAP)
             print(
                 bcolors.OKGREEN
-                + "\n * Finished epoch {:3d}  mAP: {:5.1%}  best: {:5.1%}{}\n".format(
+                + "\n * Finished epoch {:3d}  current: {:5.1%}  best: {:5.1%}{}\n".format(
```
Comment (yxgeee) [inline openunreid/apis/runner.py:317]: "current" here is not so good, please specify whether the metric is mAP or CMC here.

File: openunreid/apis/test.py [hunk 2]
```diff
@@ -98,14 +97,10 @@ def test_reid(
             # dist_gg = build_dist(cfg, gallery_features, gallery_features)
             # final_dist = re_ranking_cpu(dist, dist_qq, dist_gg)
 
-            cmc, map = evaluate_rank(final_dist, q_pids, g_pids, q_cids, g_cids)
+            cmc, map = evaluate_rank(cfg, final_dist, q_pids, g_pids, q_cids, g_cids)
         else:
             cmc, map = np.empty(50), 0.0
 
-    end_time = time.monotonic()
-    print("Testing time: ", timedelta(seconds=end_time - start_time))
```
Comment (yxgeee) [inline openunreid/apis/test.py:106]: why delete these lines?

File: openunreid/apis/test.py [hunk 3]
```diff
@@ -207,3 +202,31 @@ def infer_gan(
     print(f"\n{sep} Finished translating {sep}\n")
 
     return
+
+
+@torch.no_grad()
+def final_test(cfg, model, cmc_topk=(1, 5, 10)):
```
Comment (yxgeee) [inline openunreid/apis/test.py:208]: The module name "final_test" is somehow confusing. Maybe you could modify the original "test_reid" into "test_reid_once" and "final_test" into "test_reid".

File: tools/MMT/main.py [hunk 4]
```diff
@@ -198,21 +199,11 @@ def main():
...
-                runner.model[idx],
-                loader,
-                query,
-                gallery,
-                dataset_name=cfg.TEST.datasets[i],
-            )
+    for idx in range(len(runner.model)):
+        print("==> Test on the no.{} model".format(idx))
+        # test_reid() on self.model[idx] will only evaluate the 'mean_net'
+        # for testing 'net', use self.model[idx].module.net
```
Comment (yxgeee) [inline tools/MMT/main.py:205]: The comments need to be modified accordingly.

File: tools/SPGAN/config.yaml [hunk 5]
```diff
@@ -50,6 +50,9 @@ TRAIN:
   datasets: {'market1501': 'trainval', 'dukemtmcreid': 'trainval'}
   unsup_dataset_indexes: [1,]
 
+  # repeated number of evaluation
+  num_repeat: 10     # 10 only for vehicleid dataset, otherwise 1
```
Comment (yxgeee) [inline tools/SPGAN/config.yaml:54]: Here should be "1" I guess?

File: tools/strong_baseline/config.yaml [hunk 6]
```diff
@@ -61,8 +61,11 @@ TRAIN:
   # datasets: {'market1501': 'trainval', 'dukemtmcreid': 'trainval'}
   unsup_dataset_indexes: [0,]
 
+  # repeated number of evaluation
+  num_repeat: 1     # 10 only for vehicleid dataset, otherwise 1
+
   epochs: 50
-  iters: 400
+  iters: 800
```
Comment (yxgeee) [inline tools/strong_baseline/config.yaml:68]: As you use market-1501 in this config, the iters should be 400.

**Comment by yxgeee** [comment]
Hi Zhiwei, thanks for your pulling requests. I have made some comments, please check and modify the requests accordingly.
